Mathlib/LinearAlgebra/Matrix/GeneralLinearGroup.lean

```diff
@@ -329,4 +331,85 @@ def planeConformalMatrix {R} [Field R] (a b : R) (hab : a ^ 2 + b ^ 2 ≠ 0) :
+  · simp only [LinearIndependent, Finsupp.total_fin_zero, ker_zero, Fintype.card_ofSubsingleton,
+    Finset.univ_eq_empty, Finset.prod_empty]
+  · have {s : { s : Fin k → Fin n → 𝔽 // LinearIndependent 𝔽 s }} :
+      Fintype (Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽)) : Set (Fin n → 𝔽)) :=
+    Fintype.ofFinite _
+    have {s : { s : Fin k → Fin n → 𝔽 // LinearIndependent 𝔽 s }} :
+        Fintype ((Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽)))ᶜ : Set (Fin n → 𝔽)) :=
+      Fintype.ofFinite _
+    have (s : { s : Fin k → Fin n → 𝔽 // LinearIndependent 𝔽 s }) :
+        Fintype.card ((Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽)))ᶜ : Set (Fin n → 𝔽)) =
+      (q) ^ n - (q) ^ k := by
+        simp only [Fintype.card_compl_set, Fintype.card_pi, Finset.prod_const, Finset.card_univ,
+          Fintype.card_fin, SetLike.coe_sort_coe,
+          card_eq_pow_finrank (K := 𝔽) (V := Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽))),
+          finrank_span_eq_card s.property]
+    simp only [Fintype.card_congr (inductiveStepEquiv n), Fintype.card_sigma,
+      Fintype.sum_congr _ _ this, Finset.sum_const, Finset.card_univ, ih (Nat.le_of_succ_le hk),
+      smul_eq_mul, mul_comm, Fin.prod_univ_succAbove _ k, Fin.natCast_eq_last, Fin.val_last,
+      Fin.succAbove_last, Fin.coe_castSucc]
+
+/-- Equivalence between `GL n F` and `n` vectors of length `n` that are linearly independent. Given
+by sending a matrix to its coloumns. -/
+noncomputable def equiv_GL_linearindependent {F : Type*} [Field F] (hn : 0 < n) :
+    GL (Fin n) F ≃ { s : Fin n → (Fin n → F) // LinearIndependent F s } where
+  toFun M := ⟨transpose M, by
+    apply linearIndependent_iff_card_eq_finrank_span.2
+    rw [Set.finrank, ← rank_eq_finrank_span_cols, rank_unit]⟩
+  invFun M := by
+    apply GeneralLinearGroup.mk'' (transpose (M.1))
+    rw [show M.1ᵀ = Basis.toMatrix (Pi.basisFun F (Fin n)) (transpose (M.1)ᵀ) by rfl,
+      transpose_transpose]
+    have : Nonempty (Fin n) := Fin.pos_iff_nonempty.1 hn
+    have hdim : Fintype.card (Fin n) = FiniteDimensional.finrank F (Fin n → F) := by
+      simp only [Fintype.card_fin, FiniteDimensional.finrank_fintype_fun_eq_card]
+    let b := basisOfLinearIndependentOfCardEqFinrank M.2 hdim
+    rw [show M = ⇑b by simp only [b, coe_basisOfLinearIndependentOfCardEqFinrank]]
+    have : Invertible ((Pi.basisFun F (Fin n)).toMatrix ⇑b) :=
+      (Pi.basisFun F (Fin n)).invertibleToMatrix b
+    exact isUnit_det_of_invertible _
+  left_inv := fun x ↦ Units.ext (ext fun i j ↦ rfl)
+  right_inv := by exact congrFun rfl
+
+theorem card_GL : Nat.card (GL (Fin n) 𝔽) = ∏ i : (Fin n), (q ^ (n) - q ^ ( i : ℕ )) := by
+  by_cases hn : n = 0
+  · rw [hn]
+    simp only [Nat.card_eq_fintype_card, Fintype.card_unique, Finset.univ_eq_empty, pow_zero,
+      Finset.prod_empty]
+  · rw [Nat.card_congr (equiv_GL_linearindependent n (Nat.pos_of_ne_zero hn))]
+    exact card_LinearInependent_subtype _ (Nat.le_refl n)
```

Comment:
```suggestion
theorem card_GL : Nat.card (GL (Fin n) 𝔽) = ∏ i : (Fin n), (q ^ (n) - q ^ ( i : ℕ )) := by
  rcases Nat.eq_zero_or_pos n with rfl | hn
  · simp [Nat.card_eq_fintype_card]
  · rw [Nat.card_congr (equiv_GL_linearindependent n hn)]
    exact card_LinearInependent_subtype _ rfl.le
```
Using `rcases` with `rfl` is a nice trick: the first cases is `n=0`, and writing `rfl` replaces `n` with `0` everywhere.

Can you add a docstring to this theorem and add it in the description at the beginning of the file? Thanks!